bee-message/src/payload/salt_declaration.rs

```diff
@@ -0,0 +1,304 @@
+// Copyright 2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! Module describing the salt declaration payload.
+
+use crate::{
+    payload::{MessagePayload, PAYLOAD_LENGTH_MAX},
+    signature::ED25519_PUBLIC_KEY_LENGTH,
+    MessagePackError, MessageUnpackError, ValidationError,
+};
+
+use bee_packable::{
+    error::{PackPrefixError, UnpackPrefixError},
+    PackError, Packable, Packer, UnpackError, Unpacker, VecPrefix,
+};
+
+use alloc::vec::Vec;
+use core::{
+    convert::{Infallible, TryInto},
+    fmt,
+};
+
+/// Maximum size of payload, minus prefix `u32` and timestamp `u64`.
+const PREFIXED_BYTES_LENGTH_MAX: usize = PAYLOAD_LENGTH_MAX - 12;
+
+/// Error encountered packing a salt declaration payload.
+#[derive(Debug)]
+#[allow(missing_docs)]
+pub enum SaltDeclarationPackError {
+    InvalidPrefix,
+}
+
+impl From<PackPrefixError<Infallible, u32>> for SaltDeclarationPackError {
+    fn from(error: PackPrefixError<Infallible, u32>) -> Self {
+        match error {
+            PackPrefixError::Packable(e) => match e {},
+            PackPrefixError::Prefix(_) => Self::InvalidPrefix,
+        }
+    }
+}
+
+impl fmt::Display for SaltDeclarationPackError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::InvalidPrefix => write!(f, "invalid prefix for salt bytes"),
+        }
+    }
+}
+
+/// Error encountered unpacking a salt declaration payload.
+#[derive(Debug)]
+#[allow(missing_docs)]
+pub enum SaltDeclarationUnpackError {
+    InvalidPrefix,
+    InvalidPrefixLength(usize),
+}
+
+impl_from_infallible!(SaltDeclarationUnpackError);
+
+impl From<UnpackPrefixError<Infallible, u32>> for SaltDeclarationUnpackError {
+    fn from(error: UnpackPrefixError<Infallible, u32>) -> Self {
+        match error {
+            UnpackPrefixError::InvalidPrefixLength(len) => Self::InvalidPrefixLength(len),
+            UnpackPrefixError::Packable(e) => match e {},
+            UnpackPrefixError::Prefix(_) => Self::InvalidPrefix,
+        }
+    }
+}
+
+impl fmt::Display for SaltDeclarationUnpackError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::InvalidPrefix => write!(f, "invalid prefix for salt bytes"),
+            Self::InvalidPrefixLength(len) => write!(f, "unpacked prefix larger than maximum specified: {}", len),
+        }
+    }
+}
+
+/// Represents a [`Salt`] used in a [`SaltDeclarationPayload`].
+#[derive(Clone, Debug, PartialEq, Eq)]
+#[cfg_attr(feature = "serde1", derive(serde::Serialize, serde::Deserialize))]
+pub struct Salt {
+    /// The value of the [`Salt`].
+    bytes: Vec<u8>,
+    /// The expiration time of the [`Salt`].
+    expiry_time: u64,
```

Comment:
interesting.....it's the first time I see this term instead of `expiration`, but I guess you know best....Maybe the docs should then also use `expiry` instead of `expiration`?

Reply:
Hmm, google says that apparently `expiry` is used more by British people than `expiration` whcih is more American, so that's probably why! They mean the same thing though -- happy to use either.

Reply:
yea, I am fine with both, too. :smile: Keep `expiry` if you prefer that, but then I suggest to use that term also in the docs.